### PR TITLE
feat: Add opentelemetry support

### PR DESCRIPTION
**Pull request by zerok**
This adds basic support for OTEL traces to grafana-build. The main idea is to have at least one top-level span per pipeline (`pipeline-<CMD-NAME>`) with further sub-spans coming in where needed. 

Supported is also the `TRACEPARENT` environment variable which is set by `otel-cli` and others. This could be used for instance within the main Drone pipeline for Grafana, wrapping the calls to the various scripts so that we have a single parent span for all these grafana-build executions in there.

Screenshot 2023-09-11 at 12 04 13

### CODE REVIEW

File: pipelines/pipeline_args.go [hunk 1]
```diff
@@ -89,3 +91,39 @@ func PipelineArgsFromContext(ctx context.Context, c cliutil.CLIContext) (Pipelin
...
+		}
+		if args.GrafanaOpts.GrafanaRef != "" {
+			attributes = append(attributes, attribute.String("grafana-ref", args.GrafanaOpts.GrafanaRef))
+		}
+		if args.GrafanaOpts.EnterpriseDir != "" {
+			attributes = append(attributes, attribute.String("enterprise-dir", args.GrafanaOpts.EnterpriseDir))
+		}
+		if args.GrafanaOpts.EnterpriseRef != "" {
+			attributes = append(attributes, attribute.String("enterprise-ref", args.GrafanaOpts.EnterpriseRef))
+		}
```
Comment (kminehart) [inline pipelines/pipeline_args.go:119]: I almost think it'd be easier to query in the future if we added these attributes even with their default zero-values.

File: cmd/main.go [hunk 2]
```diff
@@ -35,31 +39,49 @@ func PipelineAction(pf pipelines.PipelineFunc) cli.ActionFunc {
...
+			span.SetStatus(codes.Error, "failed to connect to Dagger")
 			return err
 		}
+		defer client.Close()
 
-		args, err := pipelines.PipelineArgsFromContext(c.Context, c)
+		args, err := pipelines.PipelineArgsFromContext(ctx, c)
 		if err != nil {
+			span.RecordError(err)
+			span.SetStatus(codes.Error, "failed to load arguments")
```
Comment (kminehart) [inline cmd/main.go:59]: The amount of times this is repeated makes me think we need a wrapper function, don't you think?

**Review by kminehart — APPROVED** [review]
LGTM. if you feel like addressing the two comments I made you can or you can leave them, nbd either way. Thanks for adding this! can't wait to use it :D

File: pipelines/pipeline_args.go (hunk 1 above)
Comment (zerok) [reply pipelines/pipeline_args.go]: Done :)

File: cmd/main.go (hunk 2 above)
Comment (zerok) [reply cmd/main.go]: Done :)

**Pull request merged**